signers/signer-evm/src/signer.ts

```diff
@@ -53,6 +61,42 @@ export class DefaultEvmSigner implements GenericSigner<EvmTransaction> {
     this.signer = this.provider.getSigner();
   }
 
+  static buildTx(evmTx: EvmTransaction, disableV2 = false): TransactionRequest {
+    const TO_STRING_BASE = 16;
+    let tx: TransactionRequest = {};
+    if (evmTx.from) {
+      tx = { ...tx, from: evmTx.from };
+    }
+    if (evmTx.to) {
+      tx = { ...tx, to: evmTx.to };
+    }
+    if (evmTx.data) {
+      tx = { ...tx, data: evmTx.data };
+    }
+    if (evmTx.value) {
+      tx = { ...tx, value: evmTx.value };
+    }
+    if (evmTx.nonce) {
+      tx = { ...tx, nonce: evmTx.nonce };
+    }
+    if (evmTx.gasLimit) {
+      tx = { ...tx, gasLimit: evmTx.gasLimit };
+    }
+    if (evmTx.gasPrice && disableV2) {
+      tx = {
+        ...tx,
+        gasPrice: '0x' + parseInt(evmTx.gasPrice).toString(TO_STRING_BASE),
+      };
+    }
+    if (evmTx.maxFeePerGas && !disableV2) {
+      tx = { ...tx, maxFeePerGas: evmTx.maxFeePerGas };
+    }
+    if (evmTx.maxPriorityFeePerGas && !disableV2) {
+      tx = { ...tx, maxPriorityFeePerGas: evmTx.maxPriorityFeePerGas };
+    }
```

Comment:
This is the correct logic:

```suggestion
    if (!disableV2 && evmTx.maxFeePerGas && evmTx.maxPriorityFeePerGas) {
      tx = { 
        ...tx, 
        maxFeePerGas: evmTx.maxFeePerGas, 
        maxPriorityFeePerGas: evmTx.maxPriorityFeePerGas 
      };
    } else if (evmTx.gasPrice) {
      tx = {
        ...tx,
        gasPrice: '0x' + parseInt(evmTx.gasPrice).toString(TO_STRING_BASE),
      };
    }
```